google/auth/impersonated_credentials.py

```diff
@@ -321,6 +321,47 @@ def with_quota_project(self, quota_project_id):
             iam_endpoint_override=self._iam_endpoint_override,
         )
 
+    @classmethod
+    def from_authorized_user_info(cls, source_credentials, info, scopes=None):
+        """Creates a Credentials instance from parsed authorized user info.
+
+        Args:
+            source_credentials (google.oauth2.credentials.Credentials): The source
+                credentials required to impersonate the given principal.
+            info (Mapping[str, str]): The authorized user info in Google
+                format.
+            scopes (Sequence[str]): Optional list of scopes to include in the
+                credentials.
+
+        Returns:
+            google.impersonated_credentials.Credentials: The constructed
+                credentials.
+
+        Raises:
+            ValueError: If the info is not in the expected format.
+        """
+        import re
+
+        keys_needed = {"delegates", "service_account_impersonation_url", "source_credentials"}
+        missing = keys_needed.difference(six.iterkeys(info))
+
+        if missing:
+            raise ValueError(
+                "Authorized user info was not in the expected format, missing "
+                "fields {}.".format(", ".join(missing))
+            )
+
+        iam_endpoint_parser = re.search(pattern=r"(?P<service_account>[a-zA-Z-0-9]+)@(?P<project_id>[a-zA-Z-0-9]+)",
+                                        string=info.get("service_account_impersonation_url"))
+
+        return cls(
+            source_credentials=source_credentials,
+            target_principal=iam_endpoint_parser.group("service_account"),
+            target_scopes=["https://www.googleapis.com/auth/cloud-platform"],
```

Comment:
When doing impersonation, I found from the doc that `https://www.googleapis.com/auth/cloud-platform` scope is required. 
I was wondering if I should pass the user scope from `scopes` as well here. I was a bit confused with the general concept of `default_scope` and `scope`.

Reply:
`default_scopes` are set up client libraries and `scopes` are set by users. `scopes` are supposed to override `default_scopes`. The difference only makes sense for self signed JWT. If you decide to use include these scopes, you can do `scopes_to_include = scopes if scopes else default_scopes`